File: HackerRank/30-Days-of-Code/C++/day09-Recursion.cpp
#include <iostream>

using namespace std;

    //Question Link : https://www.hackerrank.com/challenges/30-recursion/problem?isFullScreen=true

int factorial(int n) {
    if(n<=1){
        return 1;
    }
    return n*factorial(n-1);
}

int main()
{   
    int n;
    cin>>n;
    cout << factorial(n);

    return 0;
}

File: HackerRank/30-Days-of-Code/Java/day09Recursion.java
import java.util.Scanner;

public class day09Recursion {

    //Question Link : https://www.hackerrank.com/challenges/30-recursion/problem?isFullScreen=true

    public static int factorial(int n) {
    if(n<=1){
        return 1;
    }
    
    return n*factorial(n-1);

    }
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        
        int n=scan.nextInt();
        int result =factorial(n);
        System.out.println(result);
        
        scan.close();
    }

}


